lib/index.js

```diff
@@ -2340,6 +2340,64 @@ Stream.prototype.batchWithTimeOrCount = function (ms, n) {
 exposeMethod('batchWithTimeOrCount');
 
 /**
+ * A moving window over the stream which returns arrays of length <= count while skipping skip elements.
+ *
+ * @id slidingBuffer
+ * @section Streams
+ * @name Stream.slidingBuffer(count, skip)
+ * @param {Number} count - number of elements in window
+ * @param {Number} skip > 0 - number of elements to skip between windows
+ * @api public
+ *
+ * _([1, 2, 3, 4, 5]).slidingBuffer(3, 1)  // => [1, 2, 3], [2, 3, 4], [3, 4, 5], [4, 5], [4]
+ * _([1, 2, 3, 4, 5]).slidingBuffer(2, 2)  // => [1, 2], [3, 4], [5]
+ */
+
+Stream.prototype.slidingBuffer = function (count, skip) {
+    var windowed = [];
+    var missing = 0;
+
+    if (typeof count != 'number' || count <= 0) {
+        throw new Error('buffer size must be >0');
+    }
+
+    if (typeof skip != 'number' || skip <= 0) {
+        throw new Error('buffer skip must be >0');
+    }
+
+    return this.consume(function (err, x, push, next) {
+        if (err) {
+            push(err);
+            next();
+        }
+        else if (x === nil) {
+            while (windowed.length) {
+                push(null, windowed);
+                windowed = windowed.slice(skip);
+            }
+            push(null, nil);
+        }
+        else {
+            if (missing > 0) {
+                missing--;
+            }
+            else {
+                windowed.push(x);
+                if (windowed.length === count) {
+                    push(null, windowed);
```

Comment:
I thought about this some more, and I think we should be careful about pushing internal state that we want to use later. We don't want a downstream operator to mess with our state. So this should be changed to

```javascript
var nextWindowed = windowed.slice(skip);
push(null, windowed);
// Turns out the Math#max is not required.
missing = skip - count;
windowed = nextWindowed;
```

Same with the while-loop above.

The concern here is code that looks like this.

```javascript
_([1, 2, 3]).slidingBuffer(3, 1)
    .doTo(function (buf) {
        if (buf.length > 1) {
            buf[1] = 0;
        }
    })
    .toArray(_.log);
// => [[1, 0, 3], [0, 0], [0]]
// => should really be [[1, 0, 3], [2, 0], [3]]
```

Please add a test for this case as well.